utils/handles/signal.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __UTILS_HANDLES_SIGNAL_H__
+#define __UTILS_HANDLES_SIGNAL_H__
+
+/**
+ * We declare a function register_signal() to handle signals while the program
+ * executing. This function will catch some exceptional behavior with the
+ * program.
+ */
+
+#include <signal.h>
+#include <stdlib.h>
+
+typedef enum UNIVERSAL_SINGAL_NUM { null, ctrl_c } universal_singal_num_t;
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+
+#include <unistd.h>
+#define ctrl_c_signal SIGINT
+#define signal_error SIG_ERR
+
+void signal_handler_posix(int signo);
+void (*end_core)(universal_singal_num_t);
+
+static inline __sighandler_t register_signal(
+    int SIG, void (*handler_core_end)(universal_singal_num_t)) {
+  end_core = handler_core_end;
+  return signal(SIG, signal_handler_posix);
+}
+
+#elif defined(_WIN32)
+#include <Windows.h>
+
+#define ctrl_c_signal CTRL_C_EVENT
+#define signal_error FALSE
+
+BOOL signal_handler_WIN(DWORD dwCtrlType);
+void (*end_core)(universal_singal_num_t);
+
+static inline __sighandler_t register_signal(
+    int SIG, void (*handler_core_end)(universal_singal_num_t)) {
+  end_core = handler_core_end;
```

Comment:
This won't work if we want to register multiple signals

Reply:
Hi @HowJMay, did you have time to think about this ?
Don't worry if you're not available these days, just tell me and I'll take the issue.
Also, happy new year !

Reply:
sorry I was not available in the past a few months, but now I am available now!
I will push a new commit in the recent few days, and Happy New Year haha =)

Reply:
Sorry I get one problem. Do we need several different signal handlers which handles one signal respectively? Or I can use one signal handler to handle all signals?

Reply:
Oh, never saw this comment sorry. Do you still have a problem ? The UNIX way to do it is that a signal is bound to a handler but many signals can be bound to the same handler.

Reply:
thank you for answering. I will do it now.

Reply:
We'd like this feature to be part of the current sprint so I'll take the time to review it and give you some guidance if needed. Feel free to discuss this in #ciri-discussion on Discord :)

Reply:
Thank you for your kindness. 
I just implemented the signal interrupt for UNIX, but I met one problem.
I can build this new code successfully with `bazel build //ciri:ciri`
however, the execution file `ciri` can't run successfully. Did I use the wrong way to build ciri?

Reply:
I'm going to need a little bit more details. How do you run ciri and what's the error ? You can build and run ciri at the same time with the command `bazel run -- ciri`

Reply:
the command help me fixing the problem, thank you!